tutorials/cifar10-distributed.ipynb

```diff
@@ -0,0 +1,1319 @@
+        "\n",
+        "On node 0 (master node):\n",
+        "\n",
+        "```\n",
+        "python -u -m torch.distributed.launch \\\n",
+        "    --nnodes=2 \\\n",
+        "    --nproc_per_node=8 \\\n",
+        "    --node_rank=0 \\\n",
+        "    --master_addr=master --master_port=2222 --use_env \\\n",
+        "    main.py run --backend=\"nccl\"\n",
+        "```\n",
+        "\n",
+        "On node 1 (worker node):\n",
+        "\n",
+        "```\n",
+        "python -u -m torch.distributed.launch \\\n",
+        "    --nnodes=2 \\\n",
+        "    --nproc_per_node=8 \\\n",
+        "    --node_rank=1 \\\n",
+        "    --master_addr=master --master_port=2222 --use_env \\\n",
+        "    main.py run --backend=\"nccl\"\n",
+        "```\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "b1Z4JfvJJ6bt"
+      },
+      "source": [
+        "### Run with internal spawining\n",
+        "\n",
+        "On node 0:\n",
+        "```\n",
+        "python -u --nnodes=2 \\\n",
+        "    --nproc_per_node=8 \\\n",
+        "    --node_rank=0 \\\n",
+        "    --master_addr=master --master_port=2222 --use_env \\\n",
+        "    main.py run --backend=\"nccl\"\n",
+        "```\n",
+        "\n",
+        "On node 1:\n",
+        "```\n",
+        "python -u --nnodes=2 \\\n",
+        "    --nproc_per_node=8 \\\n",
+        "    --node_rank=1 \\\n",
+        "    --master_addr=master --master_port=2222 --use_env \\\n",
+        "    main.py run --backend=\"nccl\"\n",
+        "```"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "KPUtXHI3KG9w"
+      },
+      "source": [
+        "### Run with horovodrun\n",
+        "\n",
+        "`np` below is calculated by `nnodes` x `nproc_per_node`.\n",
+        "\n",
+        "```\n",
+        "horovodrun -np 16 -H hostname1:8,hostname2:8 python -u main.py run --backend=\"horovod\"\n",
+        "```"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "99enig2IdNiF"
+      },
+      "source": [
+        "## Single Node, Multiple CPUs\n",
+        "\n",
+        "This is similar to Single Node, One or More GPUs. The only difference is while running the script, `backend=gloo` instead of `nccl`."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "g6s0Rn_mdewW"
+      },
+      "source": [
+        "## TPUs on Google Colab\n",
+        "\n",
+        "Go to Runtime > Change runtime type and select Hardware accelerator = TPU."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "z6e2NW3kofu8"
+      },
+      "source": [
+        "nproc_per_node = 8\n",
+        "config[\"backend\"] = \"xla-tpu\"\n",
+        "\n",
+        "with idist.Parallel(backend=config[\"backend\"], nproc_per_node=nproc_per_node) as parallel:\n",
```

Comment:
Error in Colab 

```python
import torch_xla

>>> ImportError: /usr/local/lib/python3.7/dist-packages/_XLAC.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _ZNK2at10TensorBase8data_ptrIN3c107complexIfEEEEPT_v
```

Reply:
@sdesrozis yes, we already discussed that on discord, can you help with how to patch temporarily the code such that it works with v1.9 or nightly...

Reply:
https://github.com/pytorch/xla/issues/3121

The easy fix is to downgrade pytorch to 1.8.

Reply:
Should I downgrade to PyTorch 1.8, quote this issue in the notebook and push the changes?

Reply:
Yes, let's do that and mark as temporary solution with torch_xla. Once they fixed the issue we'll remove that